packages/react/src/components/TextInput/TextInput.js

```diff
@@ -54,6 +82,8 @@ const TextInput = React.forwardRef(function TextInput(
     type,
     ref,
     className: textInputClasses,
+    readOnly,
```

Comment:
We need to add `readOnly` to `<TextArea>`, too.

Reply:
I was thinking of separating the character counter work from the read-only work unless it's fine to contain it all in this PR. if so, I will add read-only to `<TextArea>` before marking this draft as ready

Reply:
If adding read-only to `<TextArea>` would cause much code it'll make sense to separate it. Otherwise including the change in this would be fine.

Reply:
got it, just added it to vanilla and React

Reply:
we will need additional design support for readonly textarea though, I don't believe we have a spec for that